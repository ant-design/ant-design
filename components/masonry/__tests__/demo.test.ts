import demoTest from '../../../tests/shared/demoTest';

demoTest('masonry', {
  testRootProps: false,
  nameCheckPathOnly: true,
});
