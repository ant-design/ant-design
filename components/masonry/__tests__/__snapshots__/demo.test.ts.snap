// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders components/masonry/demo/DemoItem.tsx correctly 1`] = `
<div
  style="background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
/>
`;

exports[`renders components/masonry/demo/basic.tsx correctly 1`] = `
<div
  class="ant-masonry"
  data-testid="masonry-container"
  style="height:0"
>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-0"
  >
    <div
      style="height:150px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      1
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-1"
  >
    <div
      style="height:30px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      2
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-2"
  >
    <div
      style="height:90px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      3
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-3"
  >
    <div
      style="height:70px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      4
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-4"
  >
    <div
      style="height:110px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      5
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-5"
  >
    <div
      style="height:150px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      6
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-6"
  >
    <div
      style="height:130px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      7
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-7"
  >
    <div
      style="height:80px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      8
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-8"
  >
    <div
      style="height:50px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      9
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-9"
  >
    <div
      style="height:90px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      10
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-10"
  >
    <div
      style="height:100px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      11
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-11"
  >
    <div
      style="height:150px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      12
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-12"
  >
    <div
      style="height:30px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      13
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-13"
  >
    <div
      style="height:50px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      14
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-14"
  >
    <div
      style="height:80px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      15
    </div>
  </div>
</div>
`;

exports[`renders components/masonry/demo/image.tsx correctly 1`] = `
<div
  class="ant-masonry"
  data-testid="masonry-container"
  style="height:0"
>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-0"
  >
    <img
      alt="https://images.unsplash.com/photo-1510001618818-4b4e3d86bf0f"
      src="https://images.unsplash.com/photo-1510001618818-4b4e3d86bf0f?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-1"
  >
    <img
      alt="https://images.unsplash.com/photo-1507513319174-e556268bb244"
      src="https://images.unsplash.com/photo-1507513319174-e556268bb244?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-2"
  >
    <img
      alt="https://images.unsplash.com/photo-1474181487882-5abf3f0ba6c2"
      src="https://images.unsplash.com/photo-1474181487882-5abf3f0ba6c2?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-3"
  >
    <img
      alt="https://images.unsplash.com/photo-1492778297155-7be4c83960c7"
      src="https://images.unsplash.com/photo-1492778297155-7be4c83960c7?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-4"
  >
    <img
      alt="https://images.unsplash.com/photo-1508062878650-88b52897f298"
      src="https://images.unsplash.com/photo-1508062878650-88b52897f298?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-5"
  >
    <img
      alt="https://images.unsplash.com/photo-1506158278516-d720e72406fc"
      src="https://images.unsplash.com/photo-1506158278516-d720e72406fc?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-6"
  >
    <img
      alt="https://images.unsplash.com/photo-1552203274-e3c7bd771d26"
      src="https://images.unsplash.com/photo-1552203274-e3c7bd771d26?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-7"
  >
    <img
      alt="https://images.unsplash.com/photo-1528163186890-de9b86b54b51"
      src="https://images.unsplash.com/photo-1528163186890-de9b86b54b51?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-8"
  >
    <img
      alt="https://images.unsplash.com/photo-1727423304224-6d2fd99b864c"
      src="https://images.unsplash.com/photo-1727423304224-6d2fd99b864c?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-9"
  >
    <img
      alt="https://images.unsplash.com/photo-1675090391405-432434e23595"
      src="https://images.unsplash.com/photo-1675090391405-432434e23595?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-10"
  >
    <img
      alt="https://images.unsplash.com/photo-1554196967-97a8602084d9"
      src="https://images.unsplash.com/photo-1554196967-97a8602084d9?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-11"
  >
    <img
      alt="https://images.unsplash.com/photo-1491961865842-98f7befd1a60"
      src="https://images.unsplash.com/photo-1491961865842-98f7befd1a60?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-12"
  >
    <img
      alt="https://images.unsplash.com/photo-1721728613411-d56d2ddda959"
      src="https://images.unsplash.com/photo-1721728613411-d56d2ddda959?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-13"
  >
    <img
      alt="https://images.unsplash.com/photo-1731901245099-20ac7f85dbaa"
      src="https://images.unsplash.com/photo-1731901245099-20ac7f85dbaa?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-14"
  >
    <img
      alt="https://images.unsplash.com/photo-1617694455303-59af55af7e58"
      src="https://images.unsplash.com/photo-1617694455303-59af55af7e58?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-15"
  >
    <img
      alt="https://images.unsplash.com/photo-1709198165282-1dab551df890"
      src="https://images.unsplash.com/photo-1709198165282-1dab551df890?w=180&auto=format"
      style="width:100%;display:block"
    />
  </div>
</div>
`;

exports[`renders components/masonry/demo/responsive.tsx correctly 1`] = `
<div
  class="ant-masonry"
  data-testid="masonry-container"
  style="height:0"
>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-0"
  >
    <div
      style="height:120px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      1
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-1"
  >
    <div
      style="height:45px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      2
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-2"
  >
    <div
      style="height:85px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      3
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-3"
  >
    <div
      style="height:160px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      4
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-4"
  >
    <div
      style="height:95px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      5
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-5"
  >
    <div
      style="height:140px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      6
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-6"
  >
    <div
      style="height:75px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      7
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-7"
  >
    <div
      style="height:110px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      8
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-8"
  >
    <div
      style="height:65px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      9
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-9"
  >
    <div
      style="height:130px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      10
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-10"
  >
    <div
      style="height:90px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      11
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-11"
  >
    <div
      style="height:145px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      12
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-12"
  >
    <div
      style="height:55px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      13
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-13"
  >
    <div
      style="height:100px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      14
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-14"
  >
    <div
      style="height:80px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      15
    </div>
  </div>
</div>
`;

exports[`renders components/masonry/demo/sequential.tsx correctly 1`] = `
<div
  class="ant-masonry"
  data-testid="masonry-container"
  style="height:0"
>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-0"
  >
    <div
      style="height:135px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      1
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-1"
  >
    <div
      style="height:60px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      2
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-2"
  >
    <div
      style="height:105px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      3
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-3"
  >
    <div
      style="height:180px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      4
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-4"
  >
    <div
      style="height:75px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      5
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-5"
  >
    <div
      style="height:85px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      6
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-6"
  >
    <div
      style="height:115px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      7
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-7"
  >
    <div
      style="height:70px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      8
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-8"
  >
    <div
      style="height:165px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      9
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-9"
  >
    <div
      style="height:40px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      10
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-10"
  >
    <div
      style="height:120px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      11
    </div>
  </div>
  <div
    class="ant-masonry-item"
    data-ant-masonry-item="item-11"
  >
    <div
      style="height:90px;background:#f0f0f0;border-radius:4px;padding:4px;width:100%"
    >
      12
    </div>
  </div>
</div>
`;

exports[`renders components/masonry/demo/update.tsx correctly 1`] = `
Array [
  <div
    class="ant-masonry"
    data-testid="masonry-container"
    style="height:0"
  />,
  <button
    class="ant-btn ant-btn-default ant-btn-color-default ant-btn-variant-outlined"
    type="button"
  >
    <span>
      Add Item
    </span>
  </button>,
]
`;
